### PR TITLE
Merging view : switch adapter implementation

### PR DESCRIPTION
**Pull request by tadam50**
Signed-off-by: Jérémy LABOUS <jlabous@silicom.fr>
Signed-off-by: Thomas ADAM <tadam@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
New adaptated classes are implemented.
Each method adaptated not implemented yet throw an exception

**What is the current behavior?** *(You can also link to an open issue here)*
Merging view module implementation need to implement Switch adapters

**What is the new behavior (if this is a feature change)?**
Merging view module implementation step : Switch adapters implementation needed

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/SwitchAdapter.java [hunk 1]
```diff
@@ -19,46 +19,46 @@
...
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+        return getDelegate().isFictitious();
     }
 
     @Override
     public void setFictitious(final boolean fictitious) {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+        getDelegate().setFictitious(fictitious);
     }
 }
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/SwitchAdapter.java]: Is it necessary to adapt switch?

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelBusBreakerViewSwitchAdderAdapter.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ */
+package com.powsybl.iidm.mergingview;
+
+import com.powsybl.iidm.network.Switch;
+import com.powsybl.iidm.network.VoltageLevel;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+public class VoltageLevelBusBreakerViewSwitchAdderAdapter extends AbstractIdentifiableAdderAdapter<VoltageLevel.BusBreakerView.SwitchAdder> implements VoltageLevel.BusBreakerView.SwitchAdder {
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelBusBreakerViewSwitchAdderAdapter.java]: Maybe it's better to fit the IIDM API: SwitchAdder are nested classes of the relevant views

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelNodeBreakerViewSwitchAdderAdapter.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+package com.powsybl.iidm.mergingview;
+
+import com.powsybl.iidm.network.Switch;
+import com.powsybl.iidm.network.SwitchKind;
+import com.powsybl.iidm.network.VoltageLevel;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+public class VoltageLevelNodeBreakerViewSwitchAdderAdapter extends AbstractIdentifiableAdderAdapter<VoltageLevel.NodeBreakerView.SwitchAdder> implements VoltageLevel.NodeBreakerView.SwitchAdder {
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelNodeBreakerViewSwitchAdderAdapter.java]: Same remark about this class, and maybe the Voltage Level views

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelBusBreakerViewSwitchAdderAdapter.java (hunk 2 above)
Comment (tadam50) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelBusBreakerViewSwitchAdderAdapter.java]: I will make theses fields as nested classes

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelNodeBreakerViewSwitchAdderAdapter.java (hunk 3 above)
Comment (tadam50) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/VoltageLevelNodeBreakerViewSwitchAdderAdapter.java]: Ok

**Pull request merged**